corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -666,3 +667,17 @@ def get_case_types_for_domain(domain, include_deprecated=False):
         deprecated_case_types = get_data_dict_deprecated_case_types(domain)
         all_case_types -= deprecated_case_types
     return all_case_types
+
+
+def get_all_case_types_for_domain(domain):
+    """
+    Returns case types no matter if there is an existing case or not, and any defined in the data dictionary,
+    which includes those referenced in an app and those added manually.
+    """
+    data_dict_types = get_data_dict_case_types(domain)
+    module_defined_case_types = get_case_types_from_apps(domain, include_save_to_case_updates=False)
```

Comment:
Why is it necessary to exclude save to case types?

Reply:
I'm excluding it for performance reason. I plan to address it as follow up items in https://dimagi.atlassian.net/browse/SAAS-17423. Then we can treat DD as the source of truth.